app/models/grda_warehouse/data_source.rb

```diff
@@ -10,11 +10,77 @@ class GrdaWarehouse::DataSource < GrdaWarehouseBase
   has_many :organizations, class_name: GrdaWarehouse::Hud::Organization.name, inverse_of: :data_source
   has_many :projects, class_name: GrdaWarehouse::Hud::Project.name, inverse_of: :data_source
   has_many :exports, class_name: GrdaWarehouse::Hud::Export.name, inverse_of: :data_source
+  has_many :user_viewable_entities, as: :entity, class_name: 'GrdaWarehouse::Hud::UserViewableEntity'
+
   has_many :uploads
   
   scope :importable, -> { where.not(source_type: nil)}
   scope :destination, -> { where(source_type: nil)}
   scope :importable_via_samba, -> { importable.where(source_type: "samba")}
+  scope :viewable_by, -> (user) do
+    if user.roles.where( can_edit_anything_super_user: true ).exists?
```

Comment:
@dfhoughton was there a specific reason not to use: `user. can_edit_anything_super_user?` here? (and in the other two or so places?)